backend/api/Services/MissionRunService.cs

```diff
@@ -58,13 +58,28 @@ public class MissionRunService(
         ILogger<MissionRunService> logger,
         IAccessRoleService accessRoleService) : IMissionRunService
     {
+
+        private static readonly SemaphoreSlim missionRunCountSemaphore = new(1);
+
         public async Task<MissionRun> Create(MissionRun missionRun, bool triggerCreatedMissionRunEvent = true)
         {
             missionRun.Id ??= Guid.NewGuid().ToString(); // Useful for signalR messages
+
+            await missionRunCountSemaphore.WaitAsync();
+            try
+            {
+
+                missionRun.MissionRunCount = await context.MissionRuns.CountAsync() + 1;
```

Comment:
This could create issues if we ever delete a record. We have deleted records at some points when doing migrations

Reply:
The issue could be that we jump over integers in the queue but the sequence or the order will be intact I think.

Reply:
If we have missions with indexes: [1, 2, 3, 4] and then we remove the second one, then add a new mission, we'll have missions with indexed: [1, 3, 4, 4]

Reply:
Good point ! Will fix this.